src/lens.c

```diff
@@ -1556,6 +1556,8 @@ static struct prod *make_prod(struct rtn *rtn, struct lens *l) {
     ERR_BAIL(l->info);
 
     result->end->next = rtn->states;
+    if (result->end == NULL)
+        goto error;
```

Comment:
This check is not needed - though it's not entirely obvious: `add_state` calls `ERR_NOMEM` when allocation fails, which sets an error code in the `struct error` of `prod->lens->info`. `make_prod` uses `ERR_BAIL` to return early if any error has been encountered - the tricky bit is that there really is only on `struct error` in the system, so that `l->info->error` points to the same `struct error` that `prod->lens->info` points to.